### PR TITLE
Simplify fail_point usage

### PR DESCRIPTION
**Pull request by overvenus**
For pattern ($name:expr), marco fail_point can be added to any
function, though the drawback is it does not support Retrun action.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -427,32 +427,34 @@ fn set(
...
+///
+/// # fn main() { f(); my::f() }
 /// ```
-/// The full name of the `p1` fail point is `A::p1`, and `p2` is `A::M::p2`.
+/// The full name of the `p1` fail point is `A::p1`, and `p2` is `A::my::p2`.
 ///
-/// `$e` is used to transform a string to the return type of outer function or closure. If
-/// the return type is `()`, then you can use the `fail_point!($name)` shortcut.
+/// `$e` is used to transform a string to the return type of outer function or closure.
+/// If you don't need to return a customized value, then you can use the `fail_point!($name)`.
```
Comment (BusyJay) [inline src/lib.rs]: I think `return` action is used for both return early and a specified value.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/lib.rs [hunk 2]
```diff
@@ -427,32 +427,34 @@ fn set(
...
 ///
 /// If you provide an additional condition `$cond`, then the condition will be evaluated
 /// before the fail point is actually checked.
 #[macro_export]
 #[cfg(not(feature = "disabled"))]
 macro_rules! fail_point {
+    ($name:expr) => {{
+        let name = concat!(module_path!(), "::", $name);
+        $crate::eval(name, |_| {});
```
Comment (BusyJay) [inline src/lib.rs]: It should panic when `return` is specified.

File: tests/tests.rs [hunk 3]
```diff
@@ -65,21 +65,15 @@ fn test_sleep() {
 #[test]
 #[should_panic]
 fn test_panic() {
-    let f = || {
-        fail_point!("panic");
-    };
     fail::cfg("tests::panic", "panic(msg)").unwrap();
-    f()
+    fail_point!("panic");
```
Comment (BusyJay) [inline tests/tests.rs]: I think this is confusing. `fail_point` is used to define a fail point, but the usage here is just to evaluate the fail point.

File: tests/tests.rs [hunk 4]
```diff
@@ -69,12 +69,12 @@ fn test_panic() {
...
     };
     fail::cfg("tests::panic", "panic(msg)").unwrap();
-    f()
+    f();
 }
 
 #[test]
 fn test_print() {
-    let f = || {
+        let f = || {
```
Comment (BusyJay) [inline tests/tests.rs]: Wrong indent.

**Comment by BusyJay** [comment]
LGTM

File: src/lib.rs [hunk 5]
```diff
@@ -427,32 +427,37 @@ fn set(
...
 ///
 /// If you provide an additional condition `$cond`, then the condition will be evaluated
 /// before the fail point is actually checked.
 #[macro_export]
 #[cfg(not(feature = "disabled"))]
 macro_rules! fail_point {
+    ($name:expr) => {{
+        let name = concat!(module_path!(), "::", $name);
+        $crate::eval(name, |_| {
+            panic!("Return is not supported for the pattern fail_point!(\"...\")");
```
Comment (siddontang) [inline src/lib.rs]: unreachable may be better?

Comment (overvenus) [reply src/lib.rs]: I think `panic!`  is fine.

`unreachable!` is 
> A utility macro for indicating unreachable code.
> This is useful any time that the compiler can't determine that some code is unreachable.

For this pattern, actually we can reach here, but it does not support `Return`. Besides, `unreachable!` calls `panic!` internally.

**Pull request merged**